ow_power_system/src/power_system_node.cpp

```diff
@@ -109,22 +109,22 @@ int main(int argc, char* argv[]) {
         ROS_WARN_NAMED("power_system_node", "Unexpected uncertainty type for EoD prediction");
       } else { // valid prediction
         // For this example, we will print the median EoD.
-        auto samples = eod_time.getVec();
-        std::sort(samples.begin(), samples.end());
-        double eod_median = samples.at(samples.size() / 2);
+        auto samplesRUL = eod_time.getVec();
+        std::sort(samplesRUL.begin(), samplesRUL.end());
+        double eod_median = samplesRUL.at(samplesRUL.size() / 2);
         auto now =  MessageClock::now();
         auto now_s = duration_cast<std::chrono::seconds>(now.time_since_epoch());
         double rul_median = eod_median - now_s.count();
         rul_msg.data = rul_median;
       }
 
       // State of Charge Code
-      //UData currentSOC = eod_event.getState()[0];
+      UData currentSOC = eod_event.getState()[0];
       // For this example, we will print the median SOC
-      //auto samples2 = currentSOC.getVec();
-      //std::sort(samples2.begin(), samples2.end());
-      //double soc_median = samples2.at(samples2.size() / 2);
-      //soc_msg.data = soc_median;
+      auto samplesSOC = currentSOC.getVec();
+      std::sort(samplesSOC.begin(), samplesSOC.end());
+      double soc_median = samplesSOC.at(samplesSOC.size() / 2);
+      soc_msg.data = soc_median;
```

Comment:
Shouldn't these lines be within the else part of `if (eod_time.uncertainty() != UType::Samples) {`

Reply:
I am just asking if the sate_of_charge can be affected by the uncertainty

Reply:
Yes. I should have moved this code... Should be resolved now.